code/clue.js

```diff
@@ -1,59 +1,202 @@
+  weight: 8
+}
+
+const knife = {
+  name: 'Knife',
+  weight: 2
+}
+
+const candlestick = {
+  name: 'Candlestick',
+  weight: 4
+}
+
+const dumbbell = {
+  name: 'Dumbbell',
+  weight: 8,
+}
+
+const poison = {
+name: 'Poison',
+weight: 1
+}
+
+const axe = {
+  name: 'Axe',
+  weight: 12
+}
+
+const bat = {
+  name: 'Bat',
   weight: 10
 }
 
-// THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
+const trophy = {
+  name: 'Trophy',
+  weight: 8
+}
+
+const pistol = {
+  name: 'Pistol',
+  weight: 8
+}
 
-// NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
+// ARRAYS FOR ALL SUSPECTS, WEAPONS AND ROOMS
 
 const suspects = [
   mrGreen,
+  professorPlum,
+  missScarlet,
+  mrsPeacock,
+  colonelMustard,
   mrsWhite
-  // ...  and the rest
 ]
 
-const weapons = []
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol
+]
+
+const rooms = [
+  'Dining Room',
+  'Conservatory',
+  'Kitchen',
+  'Study',
+  'Library',
+  'Billiard Room',
+  'Lounge',
+  'Ballroom',
+  'Hall',
+  'Spa',
+  'Living Room',
+  'Observatory',
+  'Theater',
+  'Guest House',
+  'Patio'
+]
+
 
-const rooms = []
 
 // THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
 // YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
 const randomSelector = array => {
   return array[Math.floor(Math.random() * array.length)]
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+// AN OBJECT THAT KEEPS THE MYSTERY.
 
-// This function will be invoked when you click on the killer card.
+const mystery = {
+  killer: null,
+  weapon: null,
+  room: null
+}
+
+// This function will be invoked when you click on the KILLER CARD
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+// This will randomly select a killer from the suspects. And add that to the mystery object.
+mystery.killer = randomSelector(suspects);
+
+  document.getElementById('killerCard').style.background = mystery.killer.color;
+  document.getElementById('killerName').innerHTML = `Name: ${mystery.killer.firstName} ${mystery.killer.lastName}`;
+  document.getElementById('killerAge').innerHTML = `Age: ${mystery.killer.age}`;
+  document.getElementById('killerOccupation').innerHTML = `Occupation: ${mystery.killer.occupation}`;
+  document.getElementById('killerImage').src = mystery.killer.image;
 }
```

Comment:
I did put all these in variables, I think it made the code look very neat and became so much more readable. Like this:
`const killerCard = document.getElementById('killerCard')`
and then when I want to change the killerCard element 
`killerCard.style.background = mystery.killer.color`

Reply:
thanks, I'll have a look at your code and compare a bit, see if I understand the differences =)